src/ray/gcs/gcs_server/gcs_table_storage.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gcs_table_storage.h"
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/gcs/callback.h"
+
+namespace ray {
+namespace gcs {
+
+template <typename Key, typename Data>
+Status GcsTable<Key, Data>::Put(const JobID &job_id, const Key &key, const Data &value,
+                                const StatusCallback &callback) {
+  return store_client_->AsyncPutWithIndex(table_name_, key, job_id, value, callback);
+}
+
+template <typename Key, typename Data>
+Status GcsTable<Key, Data>::Get(const JobID &job_id, const Key &key,
+                                const OptionalItemCallback<Data> &callback) {
+  return store_client_->AsyncGet(table_name_, key, callback);
+}
+
+template <typename Key, typename Data>
+Status GcsTable<Key, Data>::GetAll(
+    const JobID &job_id, const SegmentedCallback<std::pair<Key, Data>> &callback) {
+  return store_client_->AsyncGetAll(table_name_, callback);
+}
+
+template <typename Key, typename Data>
+Status GcsTable<Key, Data>::Delete(const JobID &job_id, const Key &key,
+                                   const StatusCallback &callback) {
+  return store_client_->AsyncDelete(table_name_, key, callback);
+}
+
+template <typename Key, typename Data>
+Status GcsTable<Key, Data>::Delete(const JobID &job_id, const std::vector<Key> &keys,
```

Comment:
In what case do we need this batch delete interface?
If batch delete is needed, we should push it down to `StoreClient`, because some data base natively supports batch delete (including Redid). This would be more efficient than calling `AsyncDelete` one by one.

Reply:
fixed